test/cpp/end2end/grpc_authz_end2end_test.cc

```diff
@@ -752,7 +779,27 @@ TEST_F(GrpcAuthzEnd2EndTest, FileWatcherRecoversFromFailure) {
       "  ]"
       "}";
   grpc_core::testing::TmpFile tmp_policy(policy);
-  InitServer(CreateFileWatcherAuthzPolicyProvider(tmp_policy.name(), 1));
+  auto provider = CreateFileWatcherAuthzPolicyProvider(tmp_policy.name(), 1);
+  absl::Notification on_first_reload_done;
+  absl::Notification on_second_reload_done;
+  bool first_reload = true;
+  std::function<void(grpc_status_code code, const char* error_details)>
+      callback = [&on_first_reload_done, &on_second_reload_done, &first_reload](
```

Comment:
As mentioned elsewhere, please reset the callback after the first reload, so that we don't need all of this state machinery.

Reply:
Done.